src/App/Formatter/Post/CSV.php

```diff
@@ -67,26 +67,64 @@ public function generateHXLRows($header_rows, $hxl_rows)
         }
         return $hxl;
     }
-    private function generateHxlTagAndAttribute($key, $hxl_rows)
+
+    /**
+     * @param $form_attribute_key
+     * @param $hxl_rows
+     * @return string
+     * Iterates through all hxl attributes and looks for one that matches the given $form_attribute_key
+     */
+    private function generateHxlTagAndAttribute($form_attribute_key, $hxl_rows)
     {
         $tag = '';
         //TODO handle multi column values when CSV starts supporting them again
-
         foreach ($hxl_rows as $hxl_row) {
-            $key_set = explode('.', $key); //contains key + index of the key
-            $heading_key = isset($key_set[0]) ? $key_set[0] : null; // the heading type (sets, contact, title)
-            if ($hxl_row['key'] === $heading_key) {
-                if ($tag === '') {
-                    $tag = '#' . $hxl_row['tag_name'];
-                }
-                $attribute = $hxl_row['attribute'] && !empty($hxl_row['attribute']) ?
-                        ' +'. $hxl_row['attribute'] : '';
-                $tag = $tag.$attribute;
+            //contains key + index of the key
+            $key_set = explode('.', $form_attribute_key);
+            // the heading type (sets, contact, title)
+            $heading_key = isset($key_set[0]) ? $key_set[0] : null;
```

Comment:
Can these two lines move outside the loop?

Reply:
yup